openstack/02b-MAChigh/neighbors.c

```diff
@@ -543,7 +543,8 @@ void neighbors_updateMyDAGrankAndNeighborPreference() {
             rankIncrease = (uint16_t)(rankIncreaseIntermediary >> 10);
          }
          
-         tentativeDAGrank = neighbors_vars.neighbors[i].DAGrank+rankIncrease;
+         tentativeDAGrank = neighbors_vars.neighbors[i].DAGrank;
+         tentativeDAGrank += rankIncrease;

```

Comment:
tentativeDAGrank is uint32_t, while rankIncrease and DAGrank are uint16_t. Since MAXDAGRANK is 0xFFFF, the sum might be larger than uint16_t so the result might be smaller than 0xFFFF. When adding the summands one by one the overflow does not occur. An alternative would be 
tentativeDAGrank = neighbors_vars.neighbors[i].DAGrank+(uint32_t)rankIncrease;
Do you prefer that one?